fluentui_menus/src/main/java/com/microsoft/fluentui/tokenized/menu/Menu.kt

```diff
@@ -0,0 +1,299 @@
+                tween(
+                    durationMillis = 1,
+                    delayMillis = OutTransitionDuration - 1
+                )
+            }
+        }, label = "MenuScale"
+    ) {
+        if (it) {
+            // Menu is expanded.
+            1f
+        } else {
+            // Menu is dismissed.
+            0.8f
+        }
+    }
+
+    val alpha by transition.animateFloat(
+        transitionSpec = {
+            if (false isTransitioningTo true) {
+                // Dismissed to expanded
+                tween(durationMillis = 30)
+            } else {
+                // Expanded to dismissed.
+                tween(durationMillis = OutTransitionDuration)
+            }
+        }, label = "MenuAlpha"
+    ) {
+        if (it) {
+            // Menu is expanded.
+            1f
+        } else {
+            // Menu is dismissed.
+            0f
+        }
+    }
+    val shape = RoundedCornerShape(cornerRadius)
+    Box(
+        modifier = modifier
+            .graphicsLayer {
+                scaleX = scale
+                scaleY = scale
+                this.alpha = alpha
+                transformOrigin = transformOriginState.value
+            }
+            .shadow(elevation, shape, clip = false)
+            .background(color = color, shape = shape)
+            .clip(shape)
+            .semantics(mergeDescendants = false) {}
+            .pointerInput(Unit) {}
+            .padding(vertical = MenuVerticalPadding)
+            .width(IntrinsicSize.Max)
+    ) {
+        content()
+    }
+}
+
+/**
+ * Calculates the position of a [Menu].
+ */
+@Immutable
+internal data class MenuPositionProvider(
+    val contentOffset: DpOffset,
+    val density: Density,
+    val bottomMargin: Dp,
+    val sideMargin: Dp,
+    val onPositionCalculated: (IntRect, IntRect) -> Unit = { _, _ -> }
+) : PopupPositionProvider {
+    override fun calculatePosition(
+        anchorBounds: IntRect,
+        windowSize: IntSize,
+        layoutDirection: LayoutDirection,
+        popupContentSize: IntSize
+    ): IntOffset {
+        // The min margin below the menu, relative to the screen.
+        val bottomMarginPx = with(density) { bottomMargin.roundToPx() }
+        // The min margin to side of the menu, relative to the screen.
+        val horizontalMargin = with(density) { sideMargin.roundToPx() }
+        // The content offset specified using the dropdown offset parameter.
+        val contentOffsetX = with(density) { contentOffset.x.roundToPx() }
+        val contentOffsetY = with(density) { contentOffset.y.roundToPx() }
+
+        // Compute horizontal position.
+        val toRight = maxOf(anchorBounds.left + contentOffsetX, horizontalMargin)
+        val calculatedLeft = anchorBounds.right - contentOffsetX - popupContentSize.width
+        val toLeft =
+            if (windowSize.width - (calculatedLeft + popupContentSize.width) > horizontalMargin)
+                calculatedLeft else calculatedLeft - horizontalMargin
```

Comment:
1. should be >=
2. calculatedLeft - (extra space outside)

Reply:
simplified toLeft Calculation.